src/components/PopupConfirm.js

```diff
@@ -0,0 +1,34 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector("#popup-confirm-form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+  }
+
+  setSubmitAction(action) {
+    console.log("helpser4");
+    this._handleFormSubmit = action;
+  }
+
+  changeBtnStatus(saveStatus){
+    if (saveStatus){
+     this._submitButton.textContent = "Saving....";
+    }
+    else{
+      this._submitButton.textContent = "Save";
+    }
+  }
+
+  setEventListeners() {
+    console.log("helper5");
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+      console.log("helper7");
+      this.changeBtnStatus(true);
```

Comment:
Needs correction: You should start the loading before the api request in index.js, not here, because the submission can be handled in other way

Reply:
this.changeBtnStatus(true)  removed from line 30.  All button save statuses load before api request in index.js